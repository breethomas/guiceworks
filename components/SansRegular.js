// @flow
import React from 'react'
import css from '../styles/css'

css.fontFace({
  fontFamily: 'Helvetica-Md',
  fontStyle: 'normal',
  fontWeight: 400,
  src: 'url("/static/fonts/HelveticaNeueLTStd-Md.otf") format("opentype")',
})

const style = css({
  fontFamily: 'Helvetica-Md, "Helvetica Neue", Helvetica, sans-serif',
  fontWeight: 400,
})

type Props = {
  className?: string,
  tag?: string,
}

const SansRegular = ({ className, tag, ...props }: Props) => {
  const Component = tag || 'span'
  return (
    <Component className={`${style} ${className || ''}`} {...props} />
  )
}

SansRegular.defaultProps = {
  className: '',
  tag: 'span',
}

export default SansRegular

