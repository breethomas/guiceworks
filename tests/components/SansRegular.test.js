import React from 'react'
import renderer from 'react-test-renderer'
import SansRegular from '../../components/SansRegular'

describe('SansRegular', () => {
  let component

  beforeEach(() => {
    component = (
      <SansRegular className="SansRegularClassName" tag="p">
        SansRegular
      </SansRegular>
    )
  })

  it('shows the correct snapshot for the SansRegular component tree', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
