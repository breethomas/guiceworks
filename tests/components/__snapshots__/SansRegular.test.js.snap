// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SansRegular shows the correct snapshot for the SansRegular component tree 1`] = `
<p
  className="css-6p63zg SansRegularClassName"
>
  SansRegular
</p>
`;
